atom/browser/api/atom_api_sysinfo.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2019 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/common/native_mate_converters/file_path_converter.h"
+#include "atom/common/node_bindings.h"
+#include "atom/common/node_includes.h"
+#include "atom/common/promise_util.h"
+#include "base/system/sys_info.h"
+#include "base/threading/thread_restrictions.h"
+#include "native_mate/arguments.h"
+#include "native_mate/dictionary.h"
+
+namespace {
+
+int64_t AmountOfFreeDiskSpace(const base::FilePath& path) {
+  base::ThreadRestrictions::ScopedAllowIO allow_io;
+  return base::SysInfo::AmountOfFreeDiskSpace(path);
+}
+
+int64_t AmountOfTotalDiskSpace(const base::FilePath& path) {
+  base::ThreadRestrictions::ScopedAllowIO allow_io;
+  return base::SysInfo::AmountOfTotalDiskSpace(path);
+}
+
+double Uptime() {
+  return base::SysInfo::Uptime().InSecondsF();
+}
+
+void GetHardwareInfoCallback(const v8::Global<v8::Context>& context,
+                             scoped_refptr<atom::util::Promise> promise,
+                             base::SysInfo::HardwareInfo info) {
+  v8::Isolate* isolate = promise->isolate();
```

Comment:
This boilerplate (down to line 38) is not required, we have this logic in the promise helper now

Reply:
removed